lib/rspec/its.rb

```diff
@@ -131,7 +131,31 @@ def its(attribute, *options, &block)
           else
             attribute_chain = attribute.to_s.split('.')
             attribute_chain.inject(subject) do |inner_subject, attr|
+
+              # When NilClass:   user didnt set the rspec setting
+              # When FalseClass: skip this block
+              if RSpec.configuration.its_private_method_debug.is_a? NilClass
+
+                # respond_to? only looks in the public method space
+                # TODO Check NoMethodError
+                # TODO Check if inner_subject some kind of const
+                # inner_subject.method_defined?(attr) && !inner_subject.respond_to?(attr)
+                if !inner_subject.respond_to?(attr)
+
+                  if RSpec.configuration.its_private_method_debug
+                    # TODO if debug: show spec:line_num for better indication
+                    puts its_caller.first
+                  end
+
+                  puts "You are testing a private method. Set RSpec its_private_method_debug setting to show more info or hide it completly."
```

Comment:
You can use `RSpec.deprecate`.
I personally wouldn't want to support testing private methods implicitly, and then it would just make sense to issue a deprecation warning in 1.x, and move to `public_send` in 2.0.